vdrift: fix build, allow parallel building

lheckemann opened the pull request:
###### Motivation for this change
VDrift previously failed to build on gcc 6. This isn't fixed in any
released version (I don't think the last released version being from
2014 helps), but the latest git version at least builds.


###### Things done


- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

lheckemann committed:
vdrift: fix build, allow parallel building

VDrift previously failed to build on gcc 6. This isn't fixed in any
released version (I don't think the last released version being from
2014 helps), but the latest git version at least builds.

lheckemann committed:
vdrift: add self to maintainers

lheckemann commented:
cc @viric

lheckemann commented:
@GrahamcOfBorg build vdrift

GrahamcOfBorg reviewed (commented):
Failure for system: x86_64-darwin

```
Package ‘vdrift-git’ in /tmp/nix-ofborg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/lnl7-mac/pkgs/games/vdrift/default.nix:34 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```

GrahamcOfBorg reviewed (commented):
Failure for system: aarch64-linux

```
A    /nix/store/ix6drz1y9pq1f6kl98nrmfsg7l5lrk9i-vdrift-data-r1386/test/test1.jpk
A    /nix/store/ix6drz1y9pq1f6kl98nrmfsg7l5lrk9i-vdrift-data-r1386/test/testgoal.txt
A    /nix/store/ix6drz1y9pq1f6kl98nrmfsg7l5lrk9i-vdrift-data-r1386/test/oggtest.ogg
A    /nix/store/ix6drz1y9pq1f6kl98nrmfsg7l5lrk9i-vdrift-data-r1386/test/11k_u8_c1.wav
A    /nix/store/ix6drz1y9pq1f6kl98nrmfsg7l5lrk9i-vdrift-data-r1386/test/44k_s16_c2.wav
A    /nix/store/ix6drz1y9pq1f6kl98nrmfsg7l5lrk9i-vdrift-data-r1386/music
A    /nix/store/ix6drz1y9pq1f6kl98nrmfsg7l5lrk9i-vdrift-data-r1386/SConscript.no_data
Exported revision 1386.
cannot build derivation '/nix/store/m9c8aljysd1kz3njp0nxdy8h4kz8bhaw-vdrift-git.drv': 1 dependencies couldn't be built
error: build of '/nix/store/m9c8aljysd1kz3njp0nxdy8h4kz8bhaw-vdrift-git.drv' failed
```

GrahamcOfBorg reviewed (commented):
Success for system: x86_64-linux

```
	CPP build/pathmanager.o
Install file: "build/vdrift" as "/nix/store/182q80j2v4908mb0niz7mq327c6vd9wc-vdrift-git/bin/vdrift"
scons: done building targets.
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/182q80j2v4908mb0niz7mq327c6vd9wc-vdrift-git
shrinking /nix/store/182q80j2v4908mb0niz7mq327c6vd9wc-vdrift-git/bin/vdrift
stripping (with flags -S) in /nix/store/182q80j2v4908mb0niz7mq327c6vd9wc-vdrift-git/bin 
patching script interpreter paths in /nix/store/182q80j2v4908mb0niz7mq327c6vd9wc-vdrift-git
checking for references to /tmp/nix-build-vdrift-git.drv-0 in /nix/store/182q80j2v4908mb0niz7mq327c6vd9wc-vdrift-git...
/nix/store/182q80j2v4908mb0niz7mq327c6vd9wc-vdrift-git
```

lheckemann committed:
vdrift: split data into separate derivation

This allows vdrift to be built on hydra without taking up a large amount
of space.

lheckemann commented:
@GrahamcOfBorg build vdrift-bin

GrahamcOfBorg reviewed (commented):
Failure for system: x86_64-darwin

```
Package ‘vdrift-git’ in /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-zoidberg/pkgs/games/vdrift/default.nix:35 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```

GrahamcOfBorg reviewed (commented):
Failure for system: aarch64-linux

```
[ 31%] Linking CXX shared library libBulletCollision.so
[ 31%] Built target BulletCollision
[ 31%] Linking CXX shared library libgtest.so
[ 31%] Built target gtest
[ 31%] Linking CXX shared library libOpenGLWindow.so
[ 31%] Built target OpenGLWindow
make: *** [Makefile:141: all] Error 2
builder for '/nix/store/vxicpzqdfhnk001j6ddqnp9qyr637yj8-bullet-2.86.1.drv' failed with exit code 2
cannot build derivation '/nix/store/0p4qamlw3rls25k7rxff7zdnk0lwkxiv-vdrift-git.drv': 1 dependencies couldn't be built
error: build of '/nix/store/0p4qamlw3rls25k7rxff7zdnk0lwkxiv-vdrift-git.drv' failed
```

GrahamcOfBorg reviewed (commented):
Success for system: x86_64-linux

```
Install file: "build/vdrift" as "/nix/store/x6whf32a7vwj4lcna56bay08rgjs1a5f-vdrift-git/bin/vdrift"
scons: done building targets.
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/x6whf32a7vwj4lcna56bay08rgjs1a5f-vdrift-git
shrinking /nix/store/x6whf32a7vwj4lcna56bay08rgjs1a5f-vdrift-git/bin/vdrift
strip is /nix/store/wxn5gn8amxm1w0ikcx4gbs8a17wvss4j-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/x6whf32a7vwj4lcna56bay08rgjs1a5f-vdrift-git/bin 
patching script interpreter paths in /nix/store/x6whf32a7vwj4lcna56bay08rgjs1a5f-vdrift-git
checking for references to /tmp/nix-build-vdrift-git.drv-0 in /nix/store/x6whf32a7vwj4lcna56bay08rgjs1a5f-vdrift-git...
/nix/store/x6whf32a7vwj4lcna56bay08rgjs1a5f-vdrift-git
```

lheckemann commented:
#33494 should fix the aarch64 build.
